fastchat/serve/api_provider.py

```diff
@@ -452,3 +465,72 @@ def nvidia_api_stream_iter(model_name, messages, temp, top_p, max_tokens, api_ba
             data = json.loads(data[6:])["choices"][0]["delta"]["content"]
             text += data
             yield {"text": text, "error_code": 0}
+
+
+def cohere_api_stream_iter(
+    client_name: str,
+    model_id: str,
+    messages: list,
+    temperature: Optional[
+        float
+    ] = None,  # The SDK or API handles None for all parameters following
+    top_p: Optional[float] = None,
+    max_new_tokens: Optional[int] = None,
+    api_key: Optional[str] = None,  # default is env var CO_API_KEY
+    api_base: Optional[str] = None,
+):
+    import cohere
+
+    OPENAI_TO_COHERE_ROLE_MAP = {
+        "user": "User",
+        "assistant": "Chatbot",
+        "system": "System",
+    }
+
+    client = cohere.Client(
+        api_key=api_key or os.getenv("CO_API_KEY"),  # fern sdk doesn't accept None
```

Comment:
```suggestion
```

cohere>=5.0.1 should pick up CO_API_KEY correctly but will fix the issue with accepting None!

Reply:
sorry i'm being dumb this doesn't help you